### ISSUE TITLE
who to build

### ISSUE THREAD

**Issue by Heavy4** [Energy0124/MCFreedomLauncher#21]
who to build

**Comment by Heavy4** [comment]
how

**Comment by startjim** [comment]
use https://www.eclipse.org/

**Comment by startjim** [comment]
problem solved

**Comment by LightWayUp** [comment]
@Heavy4 Please describe details of the problem you're facing when opening issues.
For building process, you can either import the project into IDE of your choice as Maven project and compile ( see the IDE's manual for more information, as the steps varies with different IDEs ), or use the standalone Maven binary distribution that can be downloaded from Apache Maven's download page and run `mvn package` ( or any later lifecycles ) from the terminal. Refer to the Wiki for more information.

By the way, I'm reworking on the Wiki for it to provide more information and clear instructions. Still, for now, the steps described in the current Wiki should still be not too hard to follow.

**Comment by LightWayUp** [comment]
For the source code, visit the releases tab and choose "Source code (zip)" from one of the releases to download. Please note that the compiled JAR doesn't seem to run and will fail with the error "A Java exception has occurred" when using **JDK 9.0.1** ( seems to be fine with just JDK 9, however ).

**Comment by LightWayUp** [comment]
For recommended environment, import as a .idea (directory based) project with the Maven model. Recommended Maven version is 3.5.2 currently, and the recommended JDK version is 9. ( Anything below 9 will not work unless the `source` and `target` version set for Maven compiler in the POM is adjusted. ) @Heavy4

**Comment by LightWayUp** [comment]
If you face any further issues, please reply back. This issue will be closed soon in the future if all problems are solved.

**Comment by Heavy4** [comment]
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4.385 s
[INFO] Finished at: 2017-11-04T02:32:39+10:00
[INFO] Final Memory: 7M/18M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project MinecraftFreedomLauncher: Fatal error compiling: invalid target release: 9 -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

**Comment by Heavy4** [comment]
build was succeful but now error while using both 9 jdk "A Java exception has occurred" 
.....

**Comment by LightWayUp** [comment]
If the build is successful, then you're following the correct steps :).

As for the error that occurs upon running the executable JAR, I'm still investigating the cause. For this problem, please open another issue **if necessary**, although there hasn't been new releases since the last one ( which presumably still works fine as I tested it before releasing ), so I believe currently it doesn't affect too much.

**Comment by startjim** [comment]
stay on JDK 8 for now

**Comment by LightWayUp** [comment]
@Heavy4 The source has been updated and the required configuration in the POM has been added back. Please try compiling again.
(Some features are still being worked on in the master branch.)
